pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/QueryOptimizer.java

```diff
@@ -61,16 +67,24 @@ public void optimize(BrokerRequest brokerRequest, @Nullable Schema schema) {
     }
   }
 
-  /**
-   * Optimizes the given SQL query.
-   */
+  /** Optimizes the given SQL query. */
   public void optimize(PinotQuery pinotQuery, @Nullable Schema schema) {
+    optimize(pinotQuery, null, schema);
+  }
+
+  /** Optimizes the given SQL query. */
+  public void optimize(PinotQuery pinotQuery, @Nullable TableConfig config, @Nullable Schema schema) {
```

Comment:
(nit) to be more clear
```suggestion
  public void optimize(PinotQuery pinotQuery, @Nullable TableConfig tableConfig, @Nullable Schema schema) {
```

Reply:
Fixed.